### ISSUE TITLE
Change in how DELETE is handled in wp-db.php

### ISSUE THREAD

**Issue by silsbyc** [rrennick/shardb#5]
Ever since the [delete_expired_transients() function](https://developer.wordpress.org/reference/functions/delete_expired_transients/) was added in WP 4.9, we've been seeing error that resemble: 

WordPress database error Table 'wpmu_global.wp_xxxx_options' doesn't exist for query DELETE a, b FROM wp_xxxx_options a, wp_xxxx_options b
			WHERE a.option_name LIKE '\\_transient\\_%'
			AND a.option_name NOT LIKE '\\_transient\\_timeout\\_%'
			AND b.option_name = CONCAT( '_transient_timeout_', SUBSTRING( a.option_name, 12 ) )
			AND b.option_value < xxxxxxxxxxx

I worked around most of it by just disabling the automatic deletion of expired transients. But I think I may have found the source. 

In the new wp-db.php file, their quick preg_match switched syntax on line 3224 to: 
. '|DELETE(?:\s+LOW_PRIORITY|\s+QUICK|\s+IGNORE)*(?:.+?FROM)?'

Is there any reason why db.php shouldn't also switch lines 119 and 133?
. '|DELETE(?:\s+IGNORE)?(?:.+?FROM)?'
and
. '|DELETE(?:\s+LOW_PRIORITY|\s+QUICK|\s+IGNORE)*(?:.+?FROM)?'

**Comment by rrennick** [comment]
@silsbyc Thanks for the report. It does look like the issue is

`DELETE a, b ` doesn't match any of the existing regex patterns for `DELETE` so it defaults to looking for a global table.

I'll try to look at this in the next couple days.

**Comment by easaw** [comment]
@rrennick
Hi, has this been fixed?

**Comment by rrennick** [comment]
@easaw No. I haven't found the time to set up a test install. I would be happy to review a PR for it though.
